Add enhancements to the `index.md` (#73)
pydantic/pydantic-settings: docs/index.md

"""
```

1. The environment variable name is overridden using `validation_alias`. In this case, the environment variable
   `my_auth_key` will be read instead of `auth_key`.

>     Check [`Field`](TODO: Add link when available) for more information.

Reviewer: ```suggestion
    Check the [`Field` documentation](TODO: Add link when available) for more information.
```

>     Check [`Field`](TODO: Add link when available) for more information.

Reviewer: ```suggestion
    Check the [`Field` documentation](TODO: Add link when available) for more information.
```
This link might be available now? If not, it should be shortly.

2. The `AliasChoices` class allows to have multiple environment variable names for a single field.
   The first environment variable that is found will be used.

>     Check [`AliasChoices`](TODO: Add link when available) for more information.

Reviewer: ```suggestion
    Check the [`AliasChoices` documentation](TODO: Add link when available) for more information.
```

This link might also be available now?

Author: It's not.

3. The `ImportString` class allows to import an object from a string.
   In this case, the environment variable `special_function` will be read and the function `math.cos` will be imported.

4. The `env_prefix` config setting allows to set a prefix for all environment variables.

>     Check [Environment variable names](#environment-variable-names) for more information.

Reviewer: ```suggestion
    Check the [environment variable names documentation](#environment-variable-names) for more information.
```

## Environment variable names

By default, the environment variable name is the same as the field name.

You can change the prefix for all environment variables by setting the `env_prefix` config setting:

```py
from pydantic import ConfigDict

from pydantic_settings import BaseSettings

class Settings(BaseSettings):
    model_config = ConfigDict(env_prefix='my_prefix_')

    auth_key: str = 'xxx'  # will be read from `my_prefix_auth_key`
```

!!! note
    The default `env_prefix` is `''` (empty string).

If you want to change the environment variable name for a single field, you can use an alias.

There are two ways to do this:

> * Use `Field(validation_alias=...)` (see `auth_key` above)
> * Use `Field(alias=...)` (see `api_key` above)

Reviewer: ```suggestion
* Using `Field(alias=...)` (see `api_key` above)
* Using `Field(validation_alias=...)` (see `auth_key` above)
```
I changed the order since it seems to me that using `alias` would be more common.

TODO: Change the link below when available.

> Check [`Alias`] for more information about aliases.

Reviewer: This link might _also_ be available now..?

```

When `case_sensitive` is `True`, the environment variable names must match field names (optionally with a prefix),
so in this example `redis_host` could only be modified via `export redis_host`. If you want to name environment variables
all upper-case, you should name attribute all upper-case too. You can still name environment variables anything
you like through `Field(validation_alias=...)`.

!!! warning

>     In Pydantic **V1** `case_sensitive` is `False` by default and all variable names are converted to lower-case internally.

Reviewer: ```suggestion
    The config option `case_sensitive` is `False` by default and all variable names are converted to lower-case internally.
```
I think we shouldn't mention V1 any more, unless it's for the sake of contrasting v1 and v2

Author: It's supposed to be exactly that: a contrast between v1 and v2. 🤔

Reviewer: if that's the case, change  it to past tense — "In V1 case_sensitive _was_ false"

Reviewer: but the other thing that's weird is that this is pydantic-settings, so saying in pydantic-v1 is a bit weird. I'd maybe say, "In Pydantic V1, when `BaseSettings` was provided in the Pydantic library, ..."

Author: ```suggestion
    In Pydantic V1, when `BaseSettings` was provided by the Pydantic library, `case_sensitive` was `False`
    by default, and all variable names were converted to lower-case internally.
```
